### PR TITLE
[Repo] 1382 test log cleanup

### PR DESCRIPTION
**Pull request by davidicus**
related to #1382 

**Summary**

PR to fix several test log issues. 

**Change List (commits, features, bugs, etc)**

- Swapped out a couple icon names that were deprecated from Carbon for their new versions.
- Changed date format for our default Absolute date in DateTimePicker. 
- Changed test for ImageCard to test for implementation instead of looking for the console warning. Then added ignore to test since we will never trigger that if statement in our testing suite
- Added `truncateCellText` to TableCellRenderer

**Acceptance Test (how to verify the PR)**

- Go to each test file and run them. You should not see any console statements or deprecation warnings.

### CODE REVIEW

File: src/components/ImageCard/CardIcon.test.jsx [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-  afterAll(() => {
-    spy.console.mockRestore();
-    // globally this is false, but we need it true so the warning pops
-    global.__DEV__ = false;
-  });
   it('validate default', () => {
-    render(<CardIcon icon="bogus" title="title" color="#FFFFFF" />);
-    expect(spy.console).toHaveBeenCalled();
+    const { rerender } = render(<CardIcon icon="help" title="title" color="#FFFFFF" />);
+    // Help icon is the defaul used when Icon does not match list. Grabbing the path value
```
Comment (sstone2423) [inline src/components/ImageCard/CardIcon.test.jsx:9]: Small typo on `default`

**Comment by tay1orjones** [comment]
@sstone2423 We brought in `user-event` to be able to more concretely test the new combobox - specifically the [`type` event](https://github.com/testing-library/user-event#typeelement-text-options) was very helpful.

We haven't really had a convo around where we should be using it vs `fireEvent`, but it in theory sounds like a good idea to use it in most places as it "tries to simulate the real events that would happen in the browser as the user interacts with it." Explore using it and let us know. They technically consider the repo a WIP but it's already a very popular package with a solid maintenance team, including Kent himself.

File: src/components/DateTimePicker/DateTimePicker.story.jsx [hunk 2]
```diff
@@ -28,9 +28,9 @@ export const defaultRelativeValue = {
 export const defaultAbsoluteValue = {
   timeRangeKind: PICKER_KINDS.ABSOLUTE,
   timeRangeValue: {
-    startDate: '04/01/2020',
+    startDate: '2020-04-01',
     startTime: '12:34',
-    endDate: '04/06/2020',
+    endDate: '2020-04-06',
```
Comment (tay1orjones) [inline src/components/DateTimePicker/DateTimePicker.story.jsx:33]: @davidicus did you figure out that warning with the date formats, or is this okay to merge?

File: src/components/ImageCard/CardIcon.jsx [hunk 3]
```diff
@@ -16,6 +16,8 @@ const propTypes = {
 };
 
 /** This component calls out to our renderIconByName callback function with the icon name OR fails over to our legacy card icons from our legacy icon bundle */
+/* We test the implementation and do not want to trigger this console in our test logs */
+/* istanbul ignore next */
```
Comment (tay1orjones) [inline src/components/ImageCard/CardIcon.jsx:20]: Alternatively we can mock the console and expect the console to be called with a message that .includes(theWarningMessage)

One time we had the condition wrong for one of these warnings and we would've caught it if it was actually wrapped in a test.

Comment (tay1orjones) [reply src/components/ImageCard/CardIcon.jsx:20]: I see now we previously had mocked the console as I mention in the above comment and you refactored to remove and test the implementation. 

I'm fine not having a test around the `if (__DEV__ && !renderIconByName && !icons[icon]) {` condition but it's something we _can_ break and would potentially erroneously show the error. Might be worth a new issue for a broader discussion around if/when we test `warning()` usage (deprecations, errors like this, misconfigurations like in Breadcrumb with ResizeObserver polyfill, etc)

File: src/components/Table/TableBody/RowActionsCell/RowActionsCell.jsx [hunk 4]
```diff
@@ -186,6 +186,7 @@ class RowActionsCell extends React.Component {
           className={`${iotPrefix}--row-actions-container`}
         >
           <div
+            data-testid="row-action-container-background"
```
Comment (tay1orjones) [inline src/components/Table/TableBody/RowActionsCell/RowActionsCell.jsx:189]: Curious, what does this fix?

**Pull request merged**